src/gui/addnewtorrentdialog.cpp

```diff
@@ -148,16 +147,18 @@ namespace
         fsPathEdit->setCurrentIndex(existingIndex);
     }
 
-    void updatePathHistory(const QString &settingsKey, const QString &path, const int maxLength)
+    void updatePathHistory(const QString &settingsKey, const Path &path, const int maxLength)
     {
         // Add last used save path to the front of history
 
-        auto pathList = settings()->loadValue<QStringList>(settingsKey);
-        const int selectedSavePathIndex = pathList.indexOf(path);
+        auto pathList = settings()->loadValue<QStringList>(settingsKey);;
+
+        const int selectedSavePathIndex = pathList.indexOf(path.data());
```

Comment:
These data (as well as many other similar data related to the GUI state) are saved to the main configuration file only due to a very long-ago mistake that there is still no one to fix. They are not really intended for the user. If someone still climbs here, he should be scared of other things, not "non-native" paths. Why should we apply unnecessary transformations in such cases?

Reply:
IMO the main reason is consistency, that is, the users saw the paths on gui is saved the same (native separators) on file.

Reply:
> IMO the main reason is consistency, that is, the users saw the paths on gui is saved the same (native separators) on file.

Even if file is intended for internal use?

Reply:
>Even if file is intended for internal use?

Currently it is unfortunate the file is still for manual editing... If the file is truly not intended for hand editing, then it would not matter.